It now compiles, but the full derivation fails to build due to a test failure:

```
error: builder for '/nix/store/j71q2p83dzd9nhdrg80yf41rhrh3z5yn-flac-1.5.0.drv' failed with exit code 2;
       last 25 log lines:
       > OK
       > test case66: whether -o does not change input file...  \c
       > OK
       > test case67: --append --block-number=0...  \c
       > OK
       > test case67: -o --append --block-number=0...  \c
       > diff: metaflac-test-files/out.meta: No such file or directory
       > ERROR: metadata does not match expected /build/source/test/metaflac-test-files/case67-expect.meta
       >
       >  5/10 Test  #8: seeking ..........................   Passed    0.56 sec
       >  6/10 Test  #1: libFLAC ..........................   Passed    2.04 sec
       >       Start  2: libFLAC++
       >  7/10 Test #10: compression ......................   Passed    1.52 sec
       >  8/10 Test  #2: libFLAC++ ........................   Passed    0.82 sec
       >  9/10 Test  #5: flac .............................   Passed    8.45 sec
       > 10/10 Test  #9: streams ..........................   Passed  141.53 sec
       >
       > 90% tests passed, 1 tests failed out of 10
       >
       > Total Test time (real) = 142.41 sec
       >
       > The following tests FAILED:
       >          6 - metaflac (Failed)
       > Errors while running CTest
       > make: *** [Makefile:71: test] Error 8
       For full logs, run 'nix-store -l /nix/store/j71q2p83dzd9nhdrg80yf41rhrh3z5yn-flac-1.5.0.drv'.
```

It’s weird, `git log 1.4.3..1.5.0 --stat -- test/metaflac-test-files` shows no changes to the test files in 1.5.0, this `case67-expect.meta` was already there in 1.4.3 … but the tests were enabled previously, right? :thinking:

--- a/pkgs/by-name/fl/flac/package.nix
+++ b/pkgs/by-name/fl/flac/package.nix
@@ -1,8 +1,9 @@
 {
   lib,
   stdenv,
-  fetchurl,
+  fetchFromGitHub,
   cmake,
+  pandoc,
   pkg-config,
   doxygen,
   graphviz,
@@ -11,16 +12,18 @@
 
 stdenv.mkDerivation rec {
   pname = "flac";
-  version = "1.4.3";
+  version = "1.5.0";
 
-  src = fetchurl {
-    url = "http://downloads.xiph.org/releases/flac/${pname}-${version}.tar.xz";
-    # Official checksum is published at https://github.com/xiph/flac/releases/tag/${version}
-    hash = "sha256-bFjmnNIjSPRBuGEJK4JeWR0Lgi4QbebrDuTQXScgW3A=";
+  src = fetchFromGitHub {
+    owner = "xiph";
+    repo = "flac";
+    tag = version;
+    hash = "sha256-B6XRai5UOAtY/7JXNbI3YuBgazi1Xd2ZOs6vvLq9LIs=";
   };
 
   nativeBuildInputs = [
     cmake
+    pandoc
     pkg-config
     doxygen
     graphviz

--- a/pkgs/by-name/fl/flac/package.nix
+++ b/pkgs/by-name/fl/flac/package.nix
@@ -1,41 +1,39 @@
 {
-  lib,
-  stdenv,
-  fetchFromGitHub,
   cmake,
+  doxygen,
+  fetchFromGitHub,
+  graphviz,
+  lib,
+  libogg,
+  nix-update-script,
   pandoc,
   pkg-config,
-  doxygen,
-  graphviz,
-  libogg,
+  stdenv,
+  versionCheckHook,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "flac";
   version = "1.5.0";
 
   src = fetchFromGitHub {
     owner = "xiph";
     repo = "flac";
-    tag = version;
+    tag = finalAttrs.version;
     hash = "sha256-B6XRai5UOAtY/7JXNbI3YuBgazi1Xd2ZOs6vvLq9LIs=";
   };
 
   nativeBuildInputs = [
     cmake
-    pandoc
-    pkg-config
     doxygen
     graphviz
+    pandoc
+    pkg-config
   ];
 
-  buildInputs = [
-    libogg
-  ];
+  buildInputs = [ libogg ];
 
-  cmakeFlags = lib.optionals (!stdenv.hostPlatform.isStatic) [
-    "-DBUILD_SHARED_LIBS=ON"
-  ];
+  cmakeFlags = lib.optionals (!stdenv.hostPlatform.isStatic) [ "-DBUILD_SHARED_LIBS=ON" ];
 
   CFLAGS = [
     "-O3"
@@ -43,23 +41,35 @@ stdenv.mkDerivation rec {
   ];
   CXXFLAGS = [ "-O3" ];
 
-  # doCheck = true; # takes lots of time
+  patches = [ ./package.patch ];
+  doCheck = true;
 
   outputs = [
     "bin"
     "dev"
-    "out"
-    "man"
     "doc"
+    "man"
+    "out"
   ];
 
-  meta = with lib; {
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  doInstallCheck = true;
+  versionCheckProgramArg = "--version";
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
     homepage = "https://xiph.org/flac/";
     description = "Library and tools for encoding and decoding the FLAC lossless audio file format";
-    changelog = "https://xiph.org/flac/changelog.html";
+    changelog = "https://github.com/xiph/flac/releases/tag/${finalAttrs.version}";
     mainProgram = "flac";
-    platforms = platforms.all;
-    license = licenses.bsd3;
-    maintainers = with maintainers; [ ruuda ];
+    platforms = lib.platforms.all;
+    license = with lib.licenses; [
+      bsd3
+      fdl13Plus
+      gpl2Plus
+      lgpl21Plus
+    ];
+    maintainers = with lib.maintainers; [ ruuda ];
   };
-}
+})

--- a/pkgs/by-name/fl/flac/package.patch
+++ b/pkgs/by-name/fl/flac/package.patch
@@ -1,0 +1,12 @@
+diff --git a/test/test_replaygain.sh b/test/test_replaygain.sh
+index 11fb77dc..b5dea14c 100755
+--- a/test/test_replaygain.sh
++++ b/test/test_replaygain.sh
+@@ -140,7 +140,4 @@ for ACTION in $REPLAYGAIN_FREQ ; do
+   done
+ done
+
+-
+-rm -f $testdir/out.flac $testdir/out.meta
+-
+ exit 0

